### PR TITLE
Support `pod` `extern_cpp_types`

### PR DESCRIPTION
**Pull request by adetaylor**
Test for #1192

### CODE REVIEW

File: integration-tests/tests/integration_test.rs [hunk 1]
```diff
@@ -11766,6 +11766,49 @@ fn test_issue_1170() {
...
+        };"
+    };
+    run_test_ex(
+        "",
+        hdr,
+        quote! {},
+        quote! {
+
+            extern_cpp_type!("VecThingy", crate::VecThingy)
+            pod!("VecThingy")
```
Comment (RReverser) [inline integration-tests/tests/integration_test.rs:11792]: Thinking about this a bit more: maybe this test should actually be failing, *unless* the target Rust type implements `Copy`? IIUC, that's basically what `pod!` expects?

Comment (RReverser) [reply integration-tests/tests/integration_test.rs:11792]: (as in, my example from the original issue should probably also fail because I forgot `#[derive(Clopy, Copy)]` which is the only guarantee that the type is, in fact, trivial)

Comment (RReverser) [reply integration-tests/tests/integration_test.rs:11792]: Thoughts?

Comment (adetaylor) [reply integration-tests/tests/integration_test.rs:11792]: Sorry, no time for autocxx at the moment - I will reply next time I get back to the project. (However your mental model of how `pod!` works is wrong: I need to spend some time documenting exactly what are the expectations are for `extern_cpp_type!` users in the POD and non-POD case.)

**Comment by RReverser** [comment]
For now added this to my project via `{ version = "0.23.1", git = "https://github.com/google/autocxx", rev = "bd0efed" }` and can confirm this fixes my first immediate issue, thanks.

**Comment by RReverser** [comment]
Two questions. For example, I have a struct like this:

```cpp
struct RansacStats {
    size_t refinements = 0;
    size_t iterations = 0;
    size_t num_inliers = 0;
    double inlier_ratio = 0;
    double model_score = std::numeric_limits<double>::max();
};
```

I generate it with `generate!` + `pod!`, and it works great. But:

1) Why does the generated binding have `Drop`? Probably related to my misunderstanding of what `pod!` means, but it seems clearly unnecessary in this case?
2) Is it possible to have `Default` generated for such struct, where all fields actually have default initializers on the C++ side?

**Comment by RReverser** [comment]
Wait... now I see at least generated `fn new` for this struct. Not sure what changed between reruns.

**Comment by RReverser** [comment]
On some classes it still doesn't exist, but that sounds like https://github.com/google/autocxx/issues/1155.

File: integration-tests/tests/integration_test.rs (hunk 1 above)
Comment (adetaylor) [reply integration-tests/tests/integration_test.rs:11792]: I improved the documentation here in #1221.

TL;DR: deriving things in Rust is not material to whether things can be POD or non-POD; it's all about the nature of the pre-existing C++ constructors/destructors, and by the time we get as far as `#[derive]` there's no way we can add extra C++ assertions.

Overall, this `extern_cpp_type` facility was intended primarily for one set of autocxx bindings to refer to a type generated by some other autocxx/cxx bindings, i.e. cross-mod links, rather than for folks such as your good self to generate whole new fresh shiny Rust types. I've clarified that too.
